ingestors/kafka/src/main/scala/hydra/kafka/util/ConsumerGroupsOffsetConsumer.scala

```diff
@@ -0,0 +1,223 @@
+        val consumerValue = offsetMaybe.map(o => Instant.ofEpochMilli(o.commitTimestamp)).map(TopicConsumerValue.apply)
+
+        consumerKeyMaybe match {
+          case Some(consumerKey) =>
+            fs2.Stream.eval(for {
+              topicConsumer <- TopicConsumer.encode[F](consumerKey, consumerValue)
+              topicConsumerOffset <- TopicConsumerOffset.encode[F](
+                TopicConsumerOffsetKey(cr.offset.topicPartition.topic(),cr.offset.topicPartition.partition()),
+                TopicConsumerOffsetValue(cr.offset.offsetAndMetadata.offset())
+              )
+              (key, value) = topicConsumer
+              (offsetKey, offsetValue) = topicConsumerOffset
+              k <- getSerializer[F, GenericRecord](s)(isKey = true).serialize(destinationTopic, Headers.empty, key)
+              v <- getSerializer[F, GenericRecord](s)(isKey = false).serialize(destinationTopic, Headers.empty, value.orNull)
+              offsetK <- getSerializer[F, GenericRecord](s)(isKey = true).serialize(dvsInternalKafkaOffsetTopic, Headers.empty, offsetKey)
+              offsetV <- getSerializer[F, GenericRecord](s)(isKey = false).serialize(dvsInternalKafkaOffsetTopic, Headers.empty, offsetValue)
+            } yield  {
+              val p = ProducerRecord(destinationTopic, k, v)
+              val p2 = ProducerRecord(dvsInternalKafkaOffsetTopic, offsetK, offsetV)
+//              val partition = cr.offset.topicPartition.partition()
+//              myMap = myMap + (partition -> cr.offset.offsetAndMetadata.offset())
+//              val string = myMap.toList.sortBy(_._1).map(m => s"${m._1}:${m._2}").mkString(" | ")
+//              println(string)
+              ProducerRecords(List(p, p2))
+            })
+          case None =>
+            fs2.Stream.empty
+        }
+      case _ =>
+        fs2.Stream.empty
+    }).handleErrorWith { error =>
+      fs2.Stream.eval(Logger[F].warn(s"DVSConsumerStreamFailure: ${error.getMessage}")) *> fs2.Stream.empty
+    }
+  }
+
+
+  private def consumerOffsetsToInternalOffsets[F[_]: ConcurrentEffect: ContextShift: Timer: Logger]
+  (
+    sourceTopic: String,
+    destinationTopic: String,
+    bootstrapServers: String,
+    consumerGroupName: String,
+    s: SchemaRegistryClient,
+    partitionMap: PartitionOffsetMap,
+    dvsInternalKafkaOffsetTopic: String
+  ) = {
+    val settings: ConsumerSettings[F, Option[BaseKey], Option[OffsetAndMetadata]] = ConsumerSettings(
+      getConsumerGroupDeserializer[F, BaseKey](GroupMetadataManager.readMessageKey),
+      getConsumerGroupDeserializer[F, OffsetAndMetadata](GroupMetadataManager.readOffsetMessageValue)
+    )
+      .withAutoOffsetReset(AutoOffsetReset.Earliest)
+      .withBootstrapServers(bootstrapServers)
+      .withGroupId(consumerGroupName)
+
+    val producerSettings = ProducerSettings[F, Array[Byte], Array[Byte]]
+      .withBootstrapServers(bootstrapServers)
+      .withAcks(Acks.All)
```

Comment:
We have recently changed the ProducerSettings elsewhere to have a retry of 0. I would suggest that here as well.

Reply:
Will do.